extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -312,34 +346,88 @@ export default class JupyterServerInstallation {
 	}
 
 	private async installJupyterProsePackage(): Promise<void> {
+		let installJupyterCommand: string;
 		if (process.platform === constants.winPlatform) {
 			let requirements = path.join(this._pythonPackageDir, 'requirements.txt');
-			let installJupyterCommand = `"${this._pythonExecutable}" -m pip install --no-index -r "${requirements}" --find-links "${this._pythonPackageDir}" --no-warn-script-location`;
+			installJupyterCommand = `"${this._pythonExecutable}" -m pip install --no-index -r "${requirements}" --find-links "${this._pythonPackageDir}" --no-warn-script-location`;
+		}
+
+		if (installJupyterCommand) {
 			this.outputChannel.show(true);
 			this.outputChannel.appendLine(localize('msgInstallStart', "Installing required packages to run Notebooks..."));
-			await utils.executeStreamedCommand(installJupyterCommand, this.outputChannel);
+			await this.executeCommand(installJupyterCommand);
 			this.outputChannel.appendLine(localize('msgJupyterInstallDone', "... Jupyter installation complete."));
 		} else {
 			return Promise.resolve();
 		}
 	}
 
-	private async installSparkMagic(): Promise<void> {
-		if (process.platform === constants.winPlatform) {
+	private async installSparkMagic(doOnlineInstall: boolean): Promise<void> {
+		let installSparkMagic: string;
+		if (process.platform === constants.winPlatform || this._usingExistingPython) {
 			let sparkWheel = path.join(this._pythonPackageDir, `sparkmagic-${constants.sparkMagicVersion}-py3-none-any.whl`);
-			let installSparkMagic = `"${this._pythonExecutable}" -m pip install --no-index "${sparkWheel}" --find-links "${this._pythonPackageDir}" --no-warn-script-location`;
+			if (doOnlineInstall) {
+				installSparkMagic = `"${this._pythonExecutable}" -m pip install "${sparkWheel}" --no-warn-script-location`;
+			} else {
+				installSparkMagic = `"${this._pythonExecutable}" -m pip install --no-index "${sparkWheel}" --find-links "${this._pythonPackageDir}" --no-warn-script-location`;
+			}
+		}
+
+		if (installSparkMagic) {
 			this.outputChannel.show(true);
 			this.outputChannel.appendLine(localize('msgInstallingSpark', "Installing SparkMagic..."));
-			await utils.executeStreamedCommand(installSparkMagic, this.outputChannel);
-		} else {
-			return Promise.resolve();
+			await this.executeCommand(installSparkMagic);
 		}
 	}
 
+	private async installPipPackages(): Promise<void> {
+		this.outputChannel.show(true);
+		this.outputChannel.appendLine(localize('msgInstallStart', "Installing required packages to run Notebooks..."));
+
+		let installCommand = `"${this._pythonExecutable}" -m pip install jupyter pandas`;
```

Comment:
I see the string 'jupyter pandas' in a couple of places (similar to prose-codeaccelerator). Can we create separate strings for maintainability?

Reply:
Also do we need versions for these? Are we at all worried about breaking changes? I do wonder if this could mess us up.

Reply:
Some versions of Python aren't supported by the highest versions of pandas, although those versions have passed their EOL. Pandas 0.24.2 doesn't support Python 3.4 for example, but that is already past EOL. See this doc: https://devguide.python.org/devcycle/#end-of-life-branches

I'm assuming we're fine with not supporting outdated versions of Python?